addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -827,14 +827,62 @@ options.registry.OptionsTab = options.Class.extend({
-
         // The bg-XXX classes have been updated (and could be updated by another
         // option like changing color palette) -> remove the inline style that
         // was added for gray previews.
         this.$el.find(".o_we_gray_preview").each((_, e) => {
             e.style.removeProperty("background-color");
         });
+
+        // If the gray palette has been generated by Odoo standard option,
+        // the hue of all gray is the same and the saturation has been
+        // increased/decreased by the same amount for all grays in
+        // comparaison with BS grays. However the system supports any
+        // gray palette.
+
+        const hues = [];
+        const saturations = [];
+        let oneHasNoSaturation = false;
+        for (let id = 100; id <= 900; id += 100) {
+            const gray = weUtils.getCSSVariableValue(`${id}`);
+            const grayRGB = ColorpickerWidget.convertCSSColorToRgba(gray);
+            const grayHSL = ColorpickerWidget.convertRgbToHsl(grayRGB.red, grayRGB.green, grayRGB.blue);
+
+            const baseGray = weUtils.getCSSVariableValue(`base-${id}`);
+            const baseGrayRGB = ColorpickerWidget.convertCSSColorToRgba(baseGray);
+            const baseGrayHSL = ColorpickerWidget.convertRgbToHsl(baseGrayRGB.red, baseGrayRGB.green, baseGrayRGB.blue);
+
+            if (grayHSL.saturation > 0.1) {
+                if (grayHSL.lightness > 0.1 && grayHSL.lightness < 99.9) {
+                    hues.push(grayHSL.hue);
+                }
+                if (grayHSL.saturation < 99.9) {
+                    saturations.push(grayHSL.saturation - baseGrayHSL.saturation);
+                }
+            } else {
+                oneHasNoSaturation = true;
+            }
+        }
+        this.grayHueIsDefined = !!hues.length;
+
+        // Average of angles: we need to take the average of found hues
+        // because even if grays are supposed to be set to the exact
+        // same hue by the Odoo editor, there might be rounding errors
+        // during the conversion from RGB to HSL as the HSL system
+        // allows to represent more colors that the RGB hexadecimal
+        // notation. This also better support random gray palettes.
+        this.grayParams[this.GRAY_PARAMS.HUE] = (!hues.length) ? 0 : Math.round((Math.atan2(
+            hues.map(hue => Math.sin(hue * Math.PI / 180)).reduce((memo, value) => memo + value, 0) / hues.length,
+            hues.map(hue => Math.cos(hue * Math.PI / 180)).reduce((memo, value) => memo + value, 0) / hues.length
+        ) * 180 / Math.PI) + 360) % 360;
+
+        // Average of found saturation diffs, or all grays have no
+        // saturation, or all grays are fully saturated.
+        this.grayParams[this.GRAY_PARAMS.EXTRA_SATURATION] = saturations.length
+            ? saturations.reduce((memo, value) => memo + value, 0) / saturations.length
+            : (oneHasNoSaturation ? -100 : 100);
+
+        await this._super(...arguments);
```

Comment:
@oussamamessaoudi I quickly pushed a commit here with what I think we should do. Could you review the change? Like you explained, when the saturation is set to 0 for all grays, you cannot now the hue. So what I thought we should naturally be done is hiding the hue configuration when there is no saturation at all, so I did that.

Also, I noticed that the converstion from HSL to RGB when setting the gray lose some information. E.g. The base gray 500 hue is 210, I choose hue 230 but after saving the computed gray I notices its hue is not 230 but 226 :fearful: You can also see the hue setting "jump" after loading the new grays. This is normal as HSL allow to represent more colors. To mitigate that, I now take the average of all gray hues and saturation diffs.

There is still a problem: the conversion from HSL to RGB is really rounding values the bad way. Could you take a look at the `const adjustedGrayRGB = ColorpickerWidget.convertHslToRgb` line, I think the `convertHslToRgb` method could be improved to return better rounded RGB values.
E.g. Odoo `convertHslToRgb(220, 50, 95)` -> `Object { red: 235, green: 240, blue: 248 }` -> Google: 236, 240, 249 I think we really do not round the values correctly.

Reply:
I corrected the rounding value to have a result similar to the Google conversion.
IMO: 
```js       
 this.grayParams[this.GRAY_PARAMS.HUE] = (!hues.length) ? 0 : Math.round((Math.atan2(
            hues.map(hue => Math.sin(hue * Math.PI / 180)).reduce((memo, value) => memo + value, 0) / hues.length,
            hues.map(hue => Math.cos(hue * Math.PI / 180)).reduce((memo, value) => memo + value, 0) / hues.length
        ) * 180 / Math.PI) + 360) % 360;
```

- The circular average has no big effect on close values, we could use the standard average.

```js
     saturations.reduce((memo, value) => memo + value, 0) / saturations.length
```

- There is a small problem with using the average value for saturation. Choosing the max / min value returns the average value which is not the correct value.

Reply:
> The circular average has no big effect on close values, we could use the standard average.

1° and 359° should return 0° hue, not 180° which is a totally different color, so I don't know what you mean... can you explain?

> Choosing the max / min value returns the average value which is not the correct value

I don't understand what you mean, can you explain?

Reply:
About the first point:
Yes my bad I did not think in extreme cases.

About the second point:
Returning the average saturation value is not the correct value.
You can notice in the video that when choosing the maximum value in the slider, the new saturation calculation returns another value than the maximum value.

https://user-images.githubusercontent.com/18035234/113265655-aee69080-92d4-11eb-8f06-6c1b70427062.mov

Reply:
> You can notice in the video that when choosing the maximum value in the slider, the new saturation calculation returns another value than the maximum value.

Ok but that does not explain why? For the max saturation, it is not supposed to do the average anyway, the saturations array should be empty as I only consider grays which are not fully saturated. Can you debug to understand what is the bug exactly and come up with an explanation and solution? Thanks :+1:

Reply:
Indeed, I investigate and proposed a solution, the problem lay in the rounding value resulting from the conversion from rgb to hsl (returning integers instead of floating numbers).